lib/keystores.js

```diff
@@ -156,16 +157,22 @@ api.update = async ({config} = {}) => {
 
   // insert the configuration and get the updated record
   const now = Date.now();
-
-  const result = await database.collections.kmsKeystore.update({
+  const set = {
+    config,
+    'meta.updated': now
+  };
+  const query = {
     id: database.hash(config.id),
+    controller: database.hash(config.controller),
```

Comment:
This doesn't seem right. If we're looking up the database record using the new value for `controller`, we won't find it -- or the `controller` won't have changed and we have a no-op unless something else is also changing. IOW, this wouldn't allow us to change the controller. Can we add a check to the update test below to read out the record after the update to make sure the controller is actually changing? (Note we can't rely on the `InvalidStateError` exception to be thrown in this case because the sequence number always changes).

```suggestion
```

Reply:
@dlongley, you're exactly right, this is by design, note what happens inside the conditional with 'allowControllerUpdate' is true.  The line you are proposing to remove is deleted from the query object.  

If comments or some other implementation would be more clear, please advise, perhaps we can discuss at standup today.

Reply:
Oh, I see it's deleted below. I presume that it's being added to ensure that the caller doesn't apply a "partial" update by accident, i.e., they have a different controller in the config but set the flag to false and the update still works. My only suggestion then would be to only add `controller` to the query when `allowControllerUpdate` is `false` (along with a comment describing why it's there) ... rather than always adding it and then backing it out later (this is what tripped me up when reading the code).

Reply:
@dlongley that's the thing with this API, there is no such thing as a partial update at present.  I feel that secure by default is better (as currently implemented).  Would you be agreeable to a comment or two to clarify the implementation?

Reply:
I should add that the `id` property is protected by default via the same mechanism.  Changing the `id` will not turn up a record and produces the same error as an attempt to change the `controller`.

Reply:
@dlongley will you approve as is, or would you like to see your suggestion implemented?

Reply:
I can accept as-is.